sagemathinc/cocalc
# Type db schema

# Description
Adds some type checking to db-schema to ensure that get/set fields are actually declared. Remaining rules options are a work in progress.

Uses a `create` function so that you don't have to type fields and the write the fields for the object again. There could be a better way to type it but no [real luck yet](https://stackoverflow.com/questions/57517135/can-i-dynamically-type-an-object-literals-keys-to-be-based-on-a-siblings-keys).

@williamstein Maybe pull and make some typos to check how the errors look to you?

# Testing Steps
1. No compile errors
1. Run the main site
1. Check that accounts work

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
